go/worker/client/worker.go

```diff
@@ -58,8 +56,26 @@ func (w *Worker) Start() error {
 
 	// Wait for all runtimes to be initialized.
 	go func() {
-		for _, rt := range w.runtimes {
+		for id, rt := range w.runtimes {
 			<-rt.Initialized()
+
+			// After the runtime is initialized, register observer role if applicable.
+			switch config.GlobalConfig.Mode {
+			case config.ModeClient, config.ModeStatelessClient:
+				// When a node is a client node and it has an entity
+				// configured, we register it with the observer role
+				// as this may be needed for confidential runtimes.
+				rp, err := w.registration.NewRuntimeRoleProvider(node.RoleObserver, id)
```

Comment:
I think the role provider needs to be created before initialization starts, so it blocks registration until declared available.